core/domain/caching_services.py

```diff
@@ -179,7 +302,14 @@ def get_multi(namespace, sub_namespace, obj_ids):
     return result_dict
 
 
-def set_multi(namespace, sub_namespace, id_value_mapping):
+# id_value_mapping is a Dict whose values can be any of the type from
+# Exploration, Skill, Story, Topic, Collection, str. hence Any type has
+# to be used here for the value type of id_value_mapping dictionary.
+def set_multi(
+    namespace: namespace_type,
+    sub_namespace: str | None,
+    id_value_mapping: Dict[str, Any]
+) -> bool:
```

Comment:
Sorry, maybe I forgot something but cannot we use `@override` here too?

```python
@overload
def set_multi(
    namespace: Literal['collection'],
    sub_namespace: str | None, 
    id_value_mapping: Dict[str, collection_domain.Collection]
) -> bool: ...
```

and etc.?

Reply:
I was neglecting it before, because it gives only one kind of output.
but now, i thinks overload applicable here too. because of different combinations of arguments.
but i think it's not applicabe to `delete_mutli` because there is no combination of different arguments.
Done!

Reply:
Agreed, thanks!